iModelCore/Bentley/Bentley/nonport/BeFile.cpp

```diff
@@ -344,6 +344,41 @@ BeFileStatus BeFile::Write(uint32_t* bytesWritten, void const* buf, uint32_t num
 #endif
     }
 
+BeFileStatus BeFile::WriteAll(size_t* bytesWritten, void const* buf, size_t numBytes)

```

Comment:
Why? If it's supposed to write all, it should write all on all platforms.

Reply:
@pmconne my unit tests (writing > 4GB chunks on file) is failing on macOS, is it reasonable to make the function linux/windows only for now ?

Reply:
> my unit tests (writing > 4GB chunks on file) is failing on macOS

I don't see a test failure in your build log, I see a compilation error. If you were previously seeing a test failure, please provide the output.

`BeFile.cpp:347:22: error: out-of-line definition of 'WriteAll' does not match any declaration in 'BentleyM0200::BeFile'`

> is it reasonable to make the function linux/windows only for now ?

Not really.

Reply:
> > my unit tests (writing > 4GB chunks on file) is failing on macOS
> 
> I don't see a test failure in your build log, I see a compilation error. If you were previously seeing a test failure, please provide the output.
> 
> `BeFile.cpp:347:22: error: out-of-line definition of 'WriteAll' does not match any declaration in 'BentleyM0200::BeFile'`
> 
> > is it reasonable to make the function linux/windows only for now ?
> 
> Not really.

it was in a discarded run, reversing last commit so you can see the test failing

Reply:
@pmconne the failing test run https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=2825300&view=logs&j=cadecedf-f4bb-5fc8-dc3f-4e23cb5c85ca&t=84a6b01a-5161-5f48-723e-72061d45d4ab

Reply:
```
2024-04-22T18:18:12.0456790Z /Volumes/Builds/vsts_work_a/1/s/imodel-native/iModelCore/Bentley/Tests/NonPublished/BeFile_Test.cpp:341: Failure
2024-04-22T18:18:12.0456940Z Value of: status == BeFileStatus::Success
2024-04-22T18:18:12.0457060Z   Actual: false
2024-04-22T18:18:12.0457160Z Expected: true
2024-04-22T18:18:12.0457370Z Failed to write to file, file: /Volumes/Builds/vsts_work_a/1/b/MacOSARM64/static/ProductCopy/Bentley-Gtest/run/Output/Write.bin
```

All this says is that it failed in some way.
Make it print the actual BeFileStatus to understand in what specific way it failed.

Reply:
What was the problem and solution? Commit messages don't say.

Reply:
> What was the problem and solution? Commit messages don't say.

I had unix error 22: "incorrect parameters", so I figured macOS might have a check in place to make sure the correct number of bytes to set ahead of writing, I ended up capping the number of bytes to write to 0x7fff0000, and the issue went away

Reply:
Is there a reason you chose a max value of 2147418112 instead of 0x7ffff000 (2,147,479,552)?

Reply:
> Is there a reason you chose a max value of 2147418112 instead of 0x7ffff000 (2,147,479,552)?

honestly no idea where I got this number, a bad copy paste maybe, trying now with 0x7ffff000